Adds docs about SSL certificate configuration (#834)
pulp/pulpcore: docs/installation/instructions.rst

    sudo systemctl start pulpcore-worker@1
    sudo systemctl start pulpcore-worker@2

.. _ssl-setup:

SSL
---

> Pulp's REST API and content app are served over HTTP. However, users should configure HTTPS communication between

Reviewer: I think remove the first sentence since the paragraph isn't focused on that leg of the connection.

I would start with `Users should configure HTTPS ... `

clients and the reverse proxy that sits in between clients and pulpcore-api and pulpcore-content services. Pulp
installer provides three different options for configuring SSL certificates for nginx and httpd reverse proxies.

1. By default, the installer will generate a new Certificate Authority and use it to sign an SSL certificate. In
   this case, the Pulp administrator will need to distribute the Certificate Authority certificate to all clients
   that wish to communicate with Pulp. The default location for this file is ``/etc/pulp/certs/root.crt``. Clients
   will need to import this certificate to their system CA trust store.

2. If you already have an SSL Cerificate that you want to be used by the reverse proxy to encrypt communication
   with clients, Pulp Installer supports providing a path for ``pulp_webserver_ssl_cert`` and
   ``pulp_webserver_ssl_key``. The administrator is still responsible for making sure that clients trust the
   Certificate Authority that signed the SSL certificate.

> 3. Pulp installer also supports using services that use the ACME protocol to generate trusted SSL certificates.

Reviewer: ```suggestion
3. Pulp installer also supports using services that use the ACME protocol, e.g. https://letsencrypt.org/,  to generate trusted SSL certificates.
```

    sudo systemctl start pulpcore-worker@1
    sudo systemctl start pulpcore-worker@2

.. _ssl-setup:

SSL
---

> Users should configure HTTPS communication between clients and the reverse proxy that sits in between clients

Reviewer: ```suggestion
Users should configure HTTPS communication between clients and the reverse proxy that is in front of pulp services like
```

and pulpcore-api and pulpcore-content services. Pulp installer provides three different options for configuring
SSL certificates for nginx and httpd reverse proxies.

1. By default, the installer will generate a new Certificate Authority and use it to sign an SSL certificate. In
   this case, the Pulp administrator will need to distribute the Certificate Authority certificate to all clients

>    that wish to communicate with Pulp. The default location for this file is ``/etc/pulp/certs/root.crt``. Clients

Reviewer: Starting from `The default location` this could be a new paragraph if it doesn't mess with the rendered Sphinx numbering.

   will need to import this certificate to their system CA trust store. Alternatively, clients could import the
   webserver's SSL certificate. The default location for the SSL certificate is
   ``/etc/pulp/certs/pulp_webserver.crt``.

2. If you already have an SSL Cerificate that you want to be used by the reverse proxy to encrypt communication

>    with clients, Pulp Installer supports providing a path for ``pulp_webserver_tls_cert`` and

Reviewer: ```suggestion
   with clients, the Pulp Installer supports providing a path for ``pulp_webserver_tls_cert`` and
```

    sudo systemctl start pulpcore-worker@1
    sudo systemctl start pulpcore-worker@2

.. _ssl-setup:

SSL
---

Users should configure HTTPS communication between clients and the reverse proxy that is in front of pulp services

> like pulpcore-api and pulpcore-content. Pulp installer provides three different options for configuring SSL

Reviewer: nitpick I missed before.  s/Pulp installer/The Pulp Installer/

certificates for nginx and httpd reverse proxies.

1. By default, the installer will generate a new Certificate Authority and use it to sign an SSL certificate. In
   this case, the Pulp administrator will need to distribute the Certificate Authority certificate or the SSL
   certificate to all clients that wish to communicate with Pulp. Clients will need to import one of these
   certificates to their system CA trust store.

   The default location for the CA certificate is ``/etc/pulp/certs/root.crt``. The default location for the SSL
   certificate is ``/etc/pulp/certs/pulp_webserver.crt``.

2. If you already have an SSL Cerificate that you want to be used by the reverse proxy to encrypt communication
   with clients, Pulp Installer supports providing a path for ``pulp_webserver_tls_cert`` and
   ``pulp_webserver_tls_key``. The administrator is still responsible for making sure that clients trust the
   Certificate Authority that signed the SSL certificate.

> 3. Pulp installer also supports using services that use the ACME protocol, e.g. https://letsencrypt.org/,  to

Reviewer: nitpick: The Pulp Installer

>    generate trusted SSL certificates. See Pulp Installer documentation for `instructions and an example playbook

Author: What about here?

Reviewer: Here too, the Pulp Installer needs the definite article in front of it
